### PR TITLE
Zod sync validation

### PR DESCRIPTION
**Pull request by jorisre**
Related to https://github.com/react-hook-form/resolvers/issues/88

Starting with Zod, same for others resolvers coming :)

### CODE REVIEW

File: zod/src/types.ts [hunk 1]
```diff
@@ -8,7 +8,8 @@ import type { ParseParams } from 'zod/lib/src/parser';
 
 export type Resolver = <T extends z.ZodSchema<any, any>>(
   schema: T,
-  options?: ParseParams,
+  schemaOptions?: ParseParams,
+  resolverOptions?: { mode: 'async' | 'sync' },
```
Comment (bluebill1049) [inline zod/src/types.ts:12]: to use this feature, i guess users will have to this:

`resolver: async (data, options) => await zodResolver(schema, options, 'async')`

Comment (jorisre) [reply zod/src/types.ts:12]: No they have to this:
```ts
useForm({
  resolver: zodResolver(schema, options, {mode: 'sync'})
})
```

 Or if you prefer, we can do the following:

```ts
interface Options {
  schema?: ParseParams;
  resolver?: { mode: 'async' | 'sync' }
}

export type Resolver = <T extends z.ZodSchema<any, any>>(
  schema: T,
  options?: Options
) => { ... }
```

So user will have to this:
```ts
useForm({
  resolver: zodResolver(schema, {
    resolver: { mode: 'sync' },
    schema: schemaZodOptions
  })
})
```
I think I prefer this one, what do you think ?

FYI: `async` is the default mode

Comment (bluebill1049) [reply zod/src/types.ts:12]: This looks great: 

```
useForm({
  resolver: zodResolver(schema, options, {mode: 'sync'})
})
```

**Review by bluebill1049 — APPROVED** [review]

File: zod/src/types.ts (hunk 1 above)
Comment (jorisre) [reply zod/src/types.ts:12]: `options` is optional so it could be:

```ts
useForm({
  resolver: zodResolver(schema, undefined, { mode: 'sync' })
})
```

**Pull request merged**